### ISSUE TITLE
[1.4.4] Plugin crashes with unassigned variable alert.

### ISSUE THREAD

**Issue by gostapps** [Jeredian/mks-wifi-plugin#403]
### Plugin Version

1.4.4

### Cura Version

5.6.0

### Platform

Linux GNU

### Printer

Flying Bear Ghost 6

### Reproduction steps

Plugin generates an error, saying that simage variable was used before assignment. 

### Actual results

Plugin abandones.

### Expected results

Plugin expected to work normally until slice uploads to the printer

### Checklist of files to include

- [ ] Log file

### Additional information & file uploads

Here is a patch for MKSPreview.py file that fixes the problem:

```
+++ MKSPreview.py	2024-02-20 20:19:35.684697924 +0100
@@ -114,6 +114,7 @@
def generate_preview(global_container_stack, image):
     screenshot_string = ""
+    simage = 0
     meta_data = global_container_stack.getMetaData()
     Logger.log("d", "Get current preview settings.")
     encoded = False
```

**Comment by Jeredian** [comment]
Thank you for submitting a bug report. 
I will try to reproduce it on macOS and Windows.

In the meantime, could you please provide the cura.log file with this error, it may help in reproducing the problem.
Or just log section related to the issue ( error and 20-30 lines above )

```
How to find Cura log file:
`$CURA_VERSION` is the current version of Cura that's installed.
`$HOME` is your user's home directory, e.g. /home/user.
`$USER` is your user name

- Windows:
  - %APPDATA%\cura\$CURA_VERSION\cura.log
  - C:\Users\$USER\AppData\Roaming\cura\$CURA_VERSION\cura.log
- OSX:
  - $User/Library/Application\ Support/Cura/$CURA_VERSION/cura.log
- Linux:
  - $HOME/.local/share/cura/$CURA_VERSION/cura.log
 - Inside Cura application:
   - from the application menu in Help -> Show settings folder .
   
```

![image](https://github.com/PrintMakerLab/mks-wifi-plugin/assets/9591166/b70a49ae-331f-48a4-bfc2-991551866ce2)

**Comment by Jeredian** [comment]
@gostapps ,   what settings for the previews are you using ?

**Comment by Jeredian** [comment]
Reproduced.
Settings:
![image](https://github.com/PrintMakerLab/mks-wifi-plugin/assets/9591166/5c0108a7-ade3-491b-943c-57e0f8ecba7d)
Error:
![image](https://github.com/PrintMakerLab/mks-wifi-plugin/assets/9591166/f1117d7c-0f6e-4069-be9d-f36f70caa70f)

<details>
  <summary>Log file </summary>

```
2024-02-21 10:47:03,156 - DEBUG - [MainThread] MKSWifiPlugin.MKSPreview.generate_preview [118]: Get current preview settings.
2024-02-21 10:47:03,156 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]: Exception: Unable to write to file FBG5_angle_test_thicker: local variable 'simage' referenced before assignment
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]: Traceback (most recent call last):
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Qt/Bindings/OutputDeviceManagerProxy.py", line 150, in _writeToDevice
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     device.requestWrite(nodes, file_name, limit_mimetypes, file_handler, **kwargs)
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "/Applications/UltiMaker Cura.app/Contents/MacOS/share/uranium/plugins/LocalFileOutputDevice/LocalFileOutputDevice.py", line 139, in requestWrite
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     self._performWrite(file_name, selected_type, file_handler, nodes)
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "/Applications/UltiMaker Cura.app/Contents/MacOS/share/uranium/plugins/LocalFileOutputDevice/LocalFileOutputDevice.py", line 160, in _performWrite
2024-02-21 10:47:03,157 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     self.writeStarted.emit(self)
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Signal.py", line 216, in emit
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     self.__handleEmitIndirect(*args, **kwargs)
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Signal.py", line 323, in __handleEmitIndirect
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     self.__performEmit(*args, **kwargs)
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Signal.py", line 361, in __performEmit
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     signal.emit(*args, **kwargs)
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Signal.py", line 216, in emit
2024-02-21 10:47:03,158 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     self.__handleEmitIndirect(*args, **kwargs)
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Signal.py", line 323, in __handleEmitIndirect
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     self.__performEmit(*args, **kwargs)
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "UM/Signal.py", line 353, in __performEmit
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     func(*args, **kwargs)
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "/Users/jeredian/Library/Application Support/cura/5.5/plugins/MKSWifiPlugin/MKSWifiPlugin/MKSPreview.py", line 159, in add_preview
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     simage, gimage, screenshot_string = generate_preview(global_container_stack, image)
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:   File "/Users/jeredian/Library/Application Support/cura/5.5/plugins/MKSWifiPlugin/MKSWifiPlugin/MKSPreview.py", line 132, in generate_preview
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]:     return simage,gimage,screenshot_string
2024-02-21 10:47:03,159 - ERROR - [MainThread] UM.Qt.Bindings.OutputDeviceManagerProxy._writeToDevice [163]: UnboundLocalError: local variable 'simage' referenced before assignment
```
</details>

**Comment by Jeredian** [comment]
@gostapps , a pre-release with fixes is available: [1.4.5-rc](https://github.com/PrintMakerLab/mks-wifi-plugin/releases/tag/1.4.5-rc)
